langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/RegexUtil.java

```diff
@@ -61,11 +61,21 @@ static Matcher getMatcher(BRegexpValue regexpVal, BString inputStr) {
     static Matcher getMatcher(BRegexpValue regexpVal, String inputStr) {
         // Map the required ballerina regexp constructs to java.
         RegExpValue translatedRegExpVal = RegExpFactory.translateRegExpConstructs((RegExpValue) regexpVal);
-        String patternStr = StringUtils.getStringValue(translatedRegExpVal, null);
+        String patternStr = getRegexpStrValue(translatedRegExpVal);
         Pattern pattern = Pattern.compile(patternStr);
         return pattern.matcher(inputStr);
     }
 
+    private static String getRegexpStrValue(RegExpValue translatedRegExpVal) {
```

Comment:
Don't do it here. Do it at `translateRegExpConstructs()`

Reply:
@pcnfernando do we need to have the `.toString()` and `.toBalString()` methods return the non-capturing group as well?

```
string:RegExp r = re `${""}`;
r.toBalString(); // output: re `(?:)`
```

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39848/commits/26145dd1c98d0a23840715c7b3dd693760c2f5bc.